lib/cosmos/io/win32_serial_driver.rb

```diff
@@ -112,8 +112,7 @@ def read
 
       loop do
         loop do
-          # Read 1 byte
-          buffer = Win32.read_file(@handle, 1)
+          buffer = Win32.read_file(@handle, @read_max_length - data.length)
```

Comment:
It just reads up to @read_max_length without exceeding it.  Much better than reading 1 byte at a time.